src/box/lua/config/applier/box_cfg.lua

```diff
@@ -341,9 +347,31 @@ local function apply(config)
         box_cfg.listen = box.NULL
     end
 
-    -- Construct box_cfg.replication.
-    if box_cfg.replication == nil then
+    -- luacheck: ignore 542 empty if branch
+    local replication_disabled = configdata:get('replication.disabled',
+        {use_default = true})
+    if replication_disabled then
+        -- Refuse to fetch data updates from other replicaset
+        -- peers if the instance is marked as disabled.
+        --
+        -- If the data is not changed with replication, it is
+        -- easier to perform a repairing work on the instance.
+        box_cfg.replication = box.NULL
+
+        -- Issue a warning to highlight the situation for the
+        -- administrator.
+        local instance_name = configdata:names().instance_name
+        local warning = ('replication.disabled = true is set for the ' ..
+            'instance %q: the data is not replicated from or to this ' ..
+            'instance'):format(instance_name)
+        config._aboard:set({type = 'warn', message = warning})
+    elseif box_cfg.replication == nil then
+        -- Construct box_cfg.replication.
         box_cfg.replication = peer_uris(configdata)
+    else
+        -- Nothing to do: the peer URIs are set using the
+        -- replication.peers option. It is already in the box_cfg
+        -- table.
```

Comment:
It is likely due to the safe startup mode (c80b121544f60740fd2605dc2e15f86c84480c59), where we can call the `applier/box_cfg` code twice for a configured leader during applying of one configuration.

My guess was that everything is OK with the alerts, because they're cleaned up before reloading. The fix should be to just set a `key` option with some unique alert identifier for the `<aboard object>:set()` call.